### PR TITLE
Replace deprecated id -> range conversion

### PR DESCRIPTION
**Pull request by KornevNikita**
As I understand - this API was never part of the specification.
Found out this test fails to compile after https://github.com/intel/llvm/commit/f98e99c9e858984b19633825c31d8379d25a806b.

### CODE REVIEW

File: tests/hierarchical/hierarchical_non_uniform_local_range.cpp [hunk 1]
```diff
@@ -92,9 +92,12 @@ template <int dim> void check_dim(util::logger &log) {
                 local_range_total>(local_range, local_range);
         cgh.parallel_for_work_group<kernel<dim>>(
             groupRange, localRange, [=](sycl::group<dim> group_pid) {
+              sycl::range<dim> r;
+              auto group_id = group_pid.get_group_id();
+              for (int i = 0; i < dim; ++i) r[i] = group_id[i];
```
Comment (keryell) [inline tests/hierarchical/hierarchical_non_uniform_local_range.cpp:97]: On the other hand having to write this ugly code means we missed our goals somewhere.
Perhaps for SYCL Next exposing a range interface for `id` and an *explicit* constructor from a range to a `range` would make it better?

**Review by keryell — APPROVED** [review]

**Comment by KornevNikita** [comment]
@bader can we merge this one?

**Comment by steffenlarsen** [comment]
@keryell - Does the Agenda label mean you would like to wait for this to get merged?

**Comment by keryell** [comment]
> Does the Agenda label mean you would like to wait for this to get merged?

Ah no, sorry for being unclear. This is something useful to add and we should discuss in a meeting soon.

**Comment by KornevNikita** [comment]
> > Does the Agenda label mean you would like to wait for this to get merged?
> 
> Ah no, sorry for being unclear. This is something useful to add and we should discuss in a meeting soon.

We're still a little confused. We may merge this PR right now, right?

**Comment by keryell** [comment]
Yes you can merge this.

**Pull request merged**